### ISSUE TITLE
ImGuiListClipper for horizontal lists

### ISSUE THREAD

**Issue by pozemka** [ocornut/imgui#7891]
### Version/Branch of Dear ImGui:

v1.91.1 WIP

### Back-ends:

imgui_impl_win32.h + imgui_impl_dx11.h

### Compiler, OS:

Windows 10 + MSVC 2022

### Details:

Hello! I am implementing large horizontal scrollable list. Something like film roll. It would be nice to not load hundreds of not visible items?

Is it possible to use `ImGuiListClipper` for horizontal lists? Or is it designed specifically for vertical?

P.S. I think I could optimize loading based on the current scroll position, available height and width of the single list item.

**Comment by GamingMinds-DanielC** [comment]
The current `ImGuiListClipper` implementation is purely vertical. The good thing is, it's a class that is more or less added on top. It is not entangled with core functionality, instead it works by manipulating underlying ImGui data. So in principle you could copy out the class and its helper functions and then adjust that copy to work horizontally. You will probably need to access some internals to make it work if you want to go this way.

**Comment by ocornut** [comment]
I fully agree it should handle horizontal clipping. I wouldn’t recommend copying the code it’s a little too much code now.

I don’t know when I would be able to make that change, the underlying change is not so big (replacing many .y with [Axis]), it is mostly a question of designing a new API for the clipper that will work for this and other desirable features.

**Comment by BowMonk** [comment]
I am also looking for the same funcionality, would like to know if there's any priority update on this :]
